climada/hazard/tc_tracks.py

```diff
@@ -263,28 +291,82 @@ def tracks_in_exp(self, exposure, buffer=1.0):
+              officially responsible agencies.
+            * "official_3h" means to include (inofficial) 3-hourly data of the officially
+              responsible agencies (whenever available).
+            If you want to restrict to the officially reported values by the officially responsible
+            agencies (`provider="official"`) without any modifications to the original official
+            data, make sure to also set `estimate_missing=False` and `interpolate_missing=False`.
+            Otherwise, gaps in the official reporting will be filled using interpolation and/or
+            statistical estimation procedures (see below).
+            If a list is given, the following logic is applied: For each storm, the variables that
+            are not reported by the first agency for this storm are taken from the next agency in
+            the list that did report this variable for this storm. For different storms, the same
+            variable might be taken from different agencies.
+            Default: ['official_3h', 'usa', 'tokyo', 'newdelhi', 'reunion', 'bom', 'nadi',
+            'wellington', 'cma', 'hko', 'ds824', 'td9636', 'td9635', 'neumann', 'mlc']
+        rescale_windspeeds : bool, optional
+            If True, all wind speeds are linearly rescaled to 1-minute sustained winds.
+            Note however that the IBTrACS documentation (Section 5.2,
+            https://www.ncdc.noaa.gov/ibtracs/pdf/IBTrACS_version4_Technical_Details.pdf) includes
+            a warning about this kind of conversion: "While a multiplicative factor can describe
+            the numerical differences, there are procedural and observational differences between
+            agencies that can change through time, which confounds the simple multiplicative
+            factor." Default: True
         storm_id : str or list of str, optional
             IBTrACS ID of the storm, e.g. 1988234N13299, [1988234N13299, 1989260N11316].
         year_range : tuple (min_year, max_year), optional
             Year range to filter track selection. Default: (1980, 2018)
         basin : str, optional
             E.g. US, SA, NI, SI, SP, WP, EP, NA. If not provided, consider all basins.
+        interpolate_missing : bool, optional
+            If True, interpolate temporal reporting gaps within a variable (such as pressure, wind
+            speed, or radius) linearly if possible.
+            For each time step with a missing value, this procedure is only able to fill in that
+            value if there are other time steps before and after this time step for which values
+            have been reported.
+            This procedure will be applied before the statistical estimations referred to
+            by `estimate_missing`. It is applied to all variables (eye position, wind speed,
+            environmental and central pressure, storm radius and radius of maximum winds).
+            Default: True
```

Comment:
Thanks @bguillod for making a clear statement out of a diffuse question I had but was not able to communicate clearly above.

Reply:
I don't get your point. But here are the answers to your questions, maybe that helps?

> Do we know which temporal resolution the output will be, and hence to which temporal resolution we interpolate to in case of missing values?

No, you don't know this in advance. The official and inofficial time steps for each track are dictated by IBTrACS. They might even be irregular (neither 3- nor 6-hourly). Since this function reads data from IBTrACS, I think it's okay to stick to those time steps. They can still be adjusted later using `equal_timestep`.

> Do we always interpolate to 3-h first, and then select only the time steps with values in lat/lon/wind/pressure?

We apply interpolation routines before selecting or discarding any time steps.

> If we do interpolate to 3h, we will not have any missing wind/pressure value to be filled in by statistical estimation, unless these are not available for that track, right?

The temporal interpolation is only able to fill in *between* records (the temporal *gaps*). The wind speed record might end at some point while pressure is recorded for later times. In that case, temporal interpolation can't fill missing values while statistical interpolation can.

> And if we take provider="official", then we interpolate on a 3-h basis but then only keep 6-h data?

No, temporal interpolation is always to whatever time steps are defined in IBTrACS for each storm. That's what is meant by this note in the docstrings:
https://github.com/CLIMADA-project/climada_python/blob/6cb1ee7114d64d8f330a7f9e216506530b9b41ed/climada/hazard/tc_tracks.py#L324-L328